x-pack/plugins/watcher/public/sections/watch_edit/components/json_watch_edit/json_watch_edit_simulate_results.tsx

```diff
@@ -39,20 +39,22 @@ export const JsonWatchEditSimulateResults = ({
 }) => {
   const { watch } = useContext(WatchContext);
 
+  const { actionModes } = executeDetails;
+
   const getTableData = () => {
     const actions = watch.watch && watch.watch.actions;
     if (executeResults && actions) {
       const actionStatuses =
         executeResults.watchStatus && executeResults.watchStatus.actionStatuses;
-      const actionModes = executeDetails.actionModes;
       return Object.keys(actions).map(actionKey => {
         const actionStatus = actionStatuses.find(status => status.id === actionKey);
+        const { state: actionState, lastExecutionReason: actionReason } = actionStatus;
```

Comment:
Just a note that originally the code guarded against `actionStatus` being undefined. Not sure if this was a deliberate change. If you want to reimplement this guard, you could change line 50 to this:

```js
const actionStatus = actionStatuses.find(status => status.id === actionKey) || {};
```